src/ucp/api/ucp.h

```diff
@@ -630,10 +630,11 @@ typedef enum {
  *
  * @return Data-type identifier.
  *
- * @note In case of partial receive, @ref ucp_dt_iov_t::buffer can be filled
- *       with any number of bytes according to its @ref ucp_dt_iov_t::length.
+ * @note In the event of partial receive, @ref ucp_dt_iov_t::buffer can be
+ *       filled with any number of bytes according to its
+ *       @ref ucp_dt_iov_t::length.
  */
-#define ucp_dt_make_iov() (UCP_DATATYPE_IOV)
+#define ucp_dt_make_iov() ((ucp_datatype_t)UCP_DATATYPE_IOV)
```

Comment:
@yosefe I think the ucp_datatype_t casting might be potential ABI break ? ucp_datatype_t is uint64_t while UCP_DATATYPE_IOV is enum, which is compiler defined (?) and potentially can be 32bit.

Reply:
i don't see how, we are not changing the type but changing a macro (constant)

Reply:
You are trying to state that you don't know how to fix this C++ working otherwise ? (I'm just trying to clarify)

Reply:
the macro ucp_dt_make_iov() should return ucp_datatype_t. Today it returns 'enum ucp_dt_type'. This is incorrect and can be caught by C++ compiler.

Reply:
What we have right now is not right. Nevertheless this does break API for exactly the same reason...

Reply:
per offline discussion - @shamisp to provide code example which can demonstrate this PR breaks ABI/API

Reply:
@shamisp ping - if there is no example why this PR breaks ABI - we should proceed and merge it

Reply:
I will run ABI/API testing tool this week.